src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationUseStyleCheck.java

```diff
@@ -464,12 +464,12 @@ private void logCommaViolation(final DetailAST ast) {
         //comma can be null if array is empty
         final DetailAST comma = rCurly.getPreviousSibling();
 
-        if (trailingArrayComma == TrailingArrayComma.ALWAYS
-            && (comma == null || comma.getType() != TokenTypes.COMMA)) {
-            log(rCurly, MSG_KEY_ANNOTATION_TRAILING_COMMA_MISSING);
+        if (trailingArrayComma == TrailingArrayComma.ALWAYS) {
+            if (comma == null || comma.getType() != TokenTypes.COMMA) {
```

Comment:
This is a minor change splitting the `A && B` into 2 separate if statements for the next change.